File: server/src/main/java/de/litexo/model/ServerFile.java
package de.litexo.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class ServerFile {

    private String path;
    private String name;
    private ServerFileType type;


    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private boolean exists = true;


    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private long created;


    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private long lastModified;
}

File: ui/src/app/api/models/server-file.ts
/* tslint:disable */
/* eslint-disable */
import { ServerFileType } from './server-file-type';
export interface ServerFile {
  created?: number;
  exists?: boolean;
  lastModified?: number;
  name?: string;
  ownerId?: string;
  ownerName?: string;
  path?: string;
  type?: ServerFileType;
}
